src/NServiceBus.Core/Unicast/Behaviors/LoadHandlersConnector.cs

```diff
@@ -20,9 +20,7 @@ public override void Invoke(LogicalMessageProcessingStageBehavior.Context contex
         {
             bool callbackInvoked;
 
-
-            // for now we cheat and pull it from the behavior context:
-            if (!context.TryGet(CallbackInvocationBehavior.CallbackInvokedKey, out callbackInvoked))
+            if (!context.TryGet("NServiceBus.CallbackInvocation.CallbackWasInvoked", out callbackInvoked))
             {
                 callbackInvoked = false;
```

Comment:
This one has always been vierd. I think we need a context.MessageWasProcessed flag that the callbacks behavior can set instead?

This way the LoadHandlersConnecor doesn't need to know about the callbacks?